app/models/user.rb

```diff
@@ -91,8 +91,12 @@ def not_admin?
     !admin?
   end
 
-  def self.for_email(email)
-    User.find_or_create_by(email_address: email.strip.downcase)
+  def self.for_email(email, client_slug = nil)
```

Comment:
I was going for backwards compatibility. 

The case here is the `or_create_by` behavior. We want `client_slug` to be explicit when it matters. I could create a new method with your proposed name and not-nil client_slug required. I just didn't want to change `for_email` since it is used in so many places as a short-cut for `User.find_by(email_address: ...)` without the `or_create_by` behavior.